src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -1557,6 +1557,22 @@ void MarkField (FieldDefinition field, in DependencyInfo reason)
 				Annotations.Mark (field, reason);
 			}
 
+			switch (reason.Kind) {
+			case DependencyKind.AccessedViaReflection:
+			case DependencyKind.DynamicDependency:
+			case DependencyKind.DynamicallyAccessedMember:
+			case DependencyKind.InteropMethodDependency:
+				if (_context.Annotations.FlowAnnotations.ShouldWarnWhenAccessedForReflection (field))
+					_context.LogWarning (
```

Comment:
Do we need a call to ShouldSuppressAnalysisWarningsForRequiresUnreferencedCode?

Reply:
Great catch - thanks a lot!